docs/source/data_ingestion_and_preparation/connectors/weaviate.md

```diff
@@ -0,0 +1,94 @@
+
+## Configuration for Weaviate
+
+*Please see the Weaviate's [installation](https://weaviate.io/developers/weaviate/installation) page for more in-depth information on installing, configuring, and running Weaviate. We specify the setup required to run a simple demo app.*
+
+We recommend running Weaviate through docker compose. The provided `compose.yml` file runs Weaviate along with a sidecar local embedding service to make querying easier.
+
+<details>
+  <summary><i>compose.yml</i></summary>
+
+  ```yaml
+version: "3.4"
+services:
+  weaviate:
+    command:
+      - --host
+      - 0.0.0.0
+      - --port
+      - "8080"
+      - --scheme
+      - http
+    image: cr.weaviate.io/semitechnologies/weaviate:1.25.0
+    ports:
+      - 8080:8080
+      - 50051:50051
+    volumes:
+      - weaviate_data:/var/lib/weaviate
+    restart: on-failure:0
+    environment:
+      QUERY_DEFAULTS_LIMIT: 25
+      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: "true"
+      PERSISTENCE_DATA_PATH: "/var/lib/weaviate"
+      DEFAULT_VECTORIZER_MODULE: "text2vec-transformers"
+      ENABLE_MODULES: "text2vec-transformers"
+      TRANSFORMERS_INFERENCE_API: http://t2v-transformers:8080
+      CLUSTER_HOSTNAME: "node1"
+  t2v-transformers:
+    image: cr.weaviate.io/semitechnologies/transformers-inference:sentence-transformers-all-MiniLM-L6-v2
+    environment:
+      ENABLE_CUDA: 0
+volumes:
+  weaviate_data:
+  ```
+
+  Note the choice of embedding model specified in the compose file.
+</details>
+
+With this you can run Weaviate with a simple `docker compose up`.
+
+## Writing to Weaviate
+
+To write a DocSet to a Weaviate Collection from Sycamore, use the `docset.write.weaviate(...)` function. The Weaviate writer takes the following arguments:
+
+- `wv_client_args`: A dictionary of arguments to pass to the Weaviate client constructor, as in an [explicit conection](https://weaviate.io/developers/weaviate/client-libraries/python#python-client-v4-explicit-connection).
+- `collection_name`: The name of the collection to write to.
+- `collection_config`: (optional) A dictionary of keyword parameters passed to the Weaviate client's [`collection.create(...)`](https://weaviate.io/developers/weaviate/client-libraries/python#instantiate-a-collection) method. A name specified here must match the `collection_name` argument.
+- `flatten_properties`: (optional, default=`False`) Whether to flatten nested property objects during the write. Weaviate can store flattened and nested properties, but will only filter and aggregate top-level properties, meaning they need to be flattened.
+- `execute`: (optional, defaul=`True`) Whether to execute this sycamore pipeline now, or return a docset to add more transforms.
+
+To write a docset to the weaviate instance run by the docker compose above, we can write the following:
```

Comment:
Won't this be in the API reference already? Best to avoid duplication if possible

Reply:
I guess the reason to have this here is that it works with the specific configuration set in the docker compose above, so the reader can better see how the things fit together

Reply:
It seems very similar to the API [example](https://github.com/aryn-ai/sycamore/blob/24f7f65401a8950db3ba89a48014f6ae68e48a3d/lib/sycamore/sycamore/writer.py#L175)? The connection parameters and vectorizer_config is the same. Let me know what you think, this is not done in the others so I think it's worth having a standardized approach to this